pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -96,6 +96,9 @@ type PrometheusConfig struct {
 	// Define ExternalLabels for prometheus
 	// +optional
 	ExternalLabels map[string]string `json:"externalLabels,omitempty"`
+	// Enable Prometheus to be used as a receiver for the Prometheus remote write protocol. Defaults to the value of `false`.
+	// +optional
+	EnableRemoteWriteReceiver bool `json:"enableRemoteWriteReceiver,omitempty"`
```

Comment:
shouldn't this be `*bool` so that `false` isn't omitted?

Reply:
Iiuc a pointer allows us to distinguish unset from `false`/empty. I don't think we need this here, do we?

Reply:
This is also consistent with [prometheus-operator's spec](https://github.com/prometheus-operator/prometheus-operator/blob/50ec1f57f19289cda613d484f3e66327bfeb3328/pkg/apis/monitoring/v1/types.go#L169).

Reply:
I tested this against kind and it actually works as expected. My understanding was that `bool` value of false is the zero value and if you applied the following 

```yaml
spec:
  prometheusConfig:
    enableRemoteWriteReceiver: false
```
The zero value of false will be omitted (just like json.Marshal does) and `kubectl get -o yaml monitoringstack foobar` will **not** print `enableRemoteWriteReceiver: false` but to the contrary, it does in fact print it. I think json marshalling should have been changed to accomodate this behaviour.